src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCISharedBuildJobQueueService.java

```diff
@@ -157,92 +219,34 @@ private void processBuild() {
 
             // after processing a build job, remove it from the processing jobs
             processingJobs.remove(buildJob.getParticipationId());
+            localProcessingJobs--;
 
-            // process next build job
-            processBuild();
-        });
-    }
+            // process next build job if node is available
+            checkAvailabilityAndProcessNextBuild();
+        }).exceptionally(ex -> {
+            log.error("Error while processing build job: " + buildJob, ex);
 
-    /**
-     * Requeue timed out build jobs only once. If a build job is still in processedJobs after the expiration time,
-     * it might be because the node crashed. Therefore, the build job is added back to the queue.
-     */
-    @Scheduled(fixedRate = 60000)
-    protected void requeueTimedOutJobs() {
+            processingJobs.remove(buildJob.getParticipationId());
+            localProcessingJobs--;
```

Comment:
is this handled in multiple threads at the same time? If so, maybe AtomicInteger would make more senes here